src/NServiceBus.Core.Tests/Msmq/MsmqSubscriptionStorageQueueTests.cs

```diff
@@ -232,6 +232,32 @@ public void Messages_with_the_same_timestamp_have_repeatedly_same_order()
             Assert.AreEqual(queue1.Messages.Single(), queue2.Messages.Single());
         }
 
+        [Test]
+        public async Task ShouldIgnoreMessageVersionOnSubscriptions()
+        {
+            var subscriptionMessage = new MsmqSubscriptionMessage
+            {
+                ArrivedTime = DateTime.UtcNow,
+                Id = Guid.NewGuid().ToString(),
+                Body = "SomeMessage, Version=1.0.0",
+                Label = "subscriberA@server1|subscriberA"
+            };
+
+            var storageQueue = new FakeStorageQueue();
+            var subscriptionStorage = new MsmqSubscriptionStorage(storageQueue);
+
+            storageQueue.Messages.Add(subscriptionMessage);
+
+            subscriptionStorage.Init();
+
+            var subscribers = await subscriptionStorage.GetSubscriberAddressesForMessage(new[]
+            {
+                new MessageType("SomeMessage", "2.0.0")
+            }, new ContextBag());
+
+            Assert.AreEqual("subscriberA", subscribers.Single().Endpoint);
```

Comment:
I actually changed it, to use a count assert (and ignored the multiple enumerations warning). Should I revert?